lib/utils.js

```diff
@@ -97,6 +97,8 @@ var Utils = module.exports = {
     return lodash.cloneDeep(obj, function (elem) {
       // Unfortunately, lodash.cloneDeep doesn't preserve Buffer.isBuffer, which we have to rely on for binary data
       if (Buffer.isBuffer(elem)) { return elem; }
+      // Preserve data-types like `fn` across clones
+      if (elem instanceof Utils.fn) { return new Utils.fn(elem.fn, elem.args); }
```

Comment:
Which way do you guys want me to take this? Happy to do either, just want to get it out so that we can start using upsert.